pkg/model/image_target.go

```diff
@@ -3,14 +3,32 @@ package model
 import (
 	"fmt"
 
-	"github.com/google/go-cmp/cmp"
-
 	"github.com/tilt-dev/tilt/internal/container"
 	"github.com/tilt-dev/tilt/internal/sliceutils"
 	"github.com/tilt-dev/tilt/pkg/apis"
 	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
 )
 
+type LiveUpdateOptions struct {
+	// LiveUpdateOnly means we don't inject the image into the Kubernetes
+	// deploy, we only live-update to the deployed object. See this issue:
+	//
+	// https://github.com/tilt-dev/tilt/issues/4577
+	//
+	// This is a hacky way to model this right now until we
+	// firm up how images work in the apiserver.
+	LiveUpdateOnly bool
+
+	// ContainerNameSelector means Live Update should select containers based
+	// on their name rather than their image.
+	//
+	// Because the spec is updated with inferred selectors from the ImageTarget
+	// in Manifest::InferLiveUpdateSelectors(), this needs to be stored
+	// independently for use there without relying on circular reading of the
+	// spec that is being updated.
+	ContainerNameSelector string
+}
```

Comment:
This felt icky - let me know if you have suggestions here.

Originally I had the `k8s_custom_deploy` partially populate the `.Spec.Selector.Kubernetes` struct and then `Manifest::InferLiveUpdateSelectors` would not unconditionally overwrite it, but that felt worse because it was creating a weird almost-circular situation 😕